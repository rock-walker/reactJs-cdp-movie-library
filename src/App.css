.App-logo {
  height: 40px;
  color: #f55263;
  text-transform: none;
  font-weight: normal;
  margin-top: 0;
}

.App-header {
  padding: 20px 10%;
  color: white;
  text-transform: uppercase;
}

.App-title {
  font-size: 1em;
}

.App-intro {
  font-size: large;
}

footer {
  position: absolute;
  bottom: 0;
  background-color: #3c3c3c;
  padding:20px 10%;
  width: 80%;
}

footer .App-logo {
  height: 0;
}