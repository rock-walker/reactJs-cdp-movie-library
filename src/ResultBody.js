import React from 'react';
import './Components.css';

class ResultBody extends Component {
	render() {
		return();
	}
}