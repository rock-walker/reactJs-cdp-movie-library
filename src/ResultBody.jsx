import React, {Component} from 'react';
import './Components.css';
import MovieItem from './MovieItem';

class ResultBody extends Component {
    render() {
        return(
            <div className="resultBody">
                <MovieItem />
                <MovieItem />
                <MovieItem />
                <MovieItem />
            </div>
        );
    }
}

export default ResultBody;