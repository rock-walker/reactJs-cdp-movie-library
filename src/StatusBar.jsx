import React, {Component} from 'react';
import './Components.css';
import ResultCount from './ResultCount';
import SortBy from './SortBy';

class StatusBar extends Component {
    render() {
        return (
            <div className="statusBar">
                <ResultCount value="7" />
                <SortBy />
            </div>
        )
    }
}

export default StatusBar;