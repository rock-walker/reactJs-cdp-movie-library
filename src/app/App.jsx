import React, { Component } from 'react'
import '../app/app.css'
import { connect } from 'react-redux'
import { moviesCacheKeys, 
         fetchPostsIfNeeded, 
         invalidateMovies,
       } from '../actions'
import { withRouter } from 'react-router-dom'

import VisibleAppHeader from '../containers/VisibleAppHeader'
import VisibleStatusBar from '../containers/VisibleStatusBar'
import VisibleMovies from '../containers/VisibleMovies'
import AppFooter from '../appFooter/AppFooter'
import ErrorBoundary from '../common/ErrorBoundary'

class App extends Component {
    componentDidMount() {
        const { dispatch } = this.props
        dispatch(fetchPostsIfNeeded())
    }

    render() {
        const { moviesCacheKeys, movies, isFetching, lastUpdated } = this.props;
        const moviesCount = movies.length;
        const isEmpty = movies.length === 0;
        return (
            <div>
                <ErrorBoundary>
                    <VisibleAppHeader />
                    <VisibleStatusBar moviesCount={moviesCount} />
                    <VisibleMovies movies={movies} 
                                lastUpdated={lastUpdated} 
                                isFetching={isFetching}
                                isEmpty={isEmpty}/>
                    <AppFooter />
                </ErrorBoundary>
            </div>
        );
    }
}

const mapStateToProps = state => {
    const { moviesCacheKeys, moviesBySearch } = state.appReducers
    const {
        isFetching,
        lastUpdated,
        items: movies
    } = moviesBySearch[moviesCacheKeys] || {
        isFetching: true,
        items: []
    }
    const { location } = state.routerReducer
    return {
        moviesCacheKeys,
        movies,
        isFetching,
        lastUpdated,
        location
    }
}

export default connect(mapStateToProps)(App)