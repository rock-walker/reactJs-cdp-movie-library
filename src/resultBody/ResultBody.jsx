import React, {Component} from 'react';
import '../resultBody/resultBody.css';
import MovieItem from '../movieItem/MovieItem';

class ResultBody extends Component {
    constructor(props){
        super(props);
    }
    render() {
        return(
            <div className="resultBody">
               {
                this.props.movies.map((item) =>
                    <MovieItem title={item.title} 
                        date={item.release_date}
                        genres={item.genres}
                        item={item} />
                    )
               }
            </div>
        );
    }
}

export default ResultBody;