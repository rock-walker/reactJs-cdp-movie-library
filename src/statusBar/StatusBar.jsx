import React, {Component} from 'react'
import '../statusBar/statusBar.css'
import {Switch, Route} from 'react-router'
import ResultCount from '../resultCount/ResultCount'
import VisibleSortBy from '../containers/VisibleSortBy'

class StatusBar extends Component {
    render() {
        const { genre, moviesCount } = this.props
        return (
            <div className="statusBar">
                <Switch>
                    <Route path="/film/:id" render = {() => 
                        <span>Films by {genre} genre</span>
                        }/>
                    <Route path="/" render = {() =>
                        <div>
                            <ResultCount value={moviesCount} />
                            <VisibleSortBy/>
                        </div>
                        }/>
                </Switch>
            </div>
        )
    }
}

export default StatusBar;