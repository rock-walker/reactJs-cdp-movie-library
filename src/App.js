import React, { Component } from 'react';
import './App.css';
import AppHeader from './AppHeader';
import StatusBar from './StatusBar';
import ResultBody from './ResultBody';
import AppFooter from './AppFooter';
import ErrorBoundary from './ErrorBoundary';

class App extends Component {
  render() {
    return (
      <div>
        <ErrorBoundary>
            <AppHeader />
            <StatusBar />
            <ResultBody/>
            <AppFooter />
        </ErrorBoundary>
      </div>
    );
  }
}

export default App;