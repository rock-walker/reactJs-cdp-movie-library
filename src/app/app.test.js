import React from 'react';
import App from '../app/App';
import StatusBar from '../statusBar/StatusBar';
import ResultBody from '../resultBody/ResultBody'
import {mount, configure} from 'enzyme';  
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

describe('App', () => {
    test('render app', () => {
        const tree = renderer.create(
                <App/>
            ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sort movies', () => {
        //unsorted array;
        const app = mount(<App />);
        expect(app.state().movies[0].title).toBe("Kill bill v.1");
        
        var statusBarChange = app.find(StatusBar).prop('OnSortingChange');
        var resultBody = app.find(ResultBody);

        //sort by rate
        statusBarChange(false);
        var title = resultBody.props().movies[0].title;
        expect(title).toBe("Pulp fiction");

        //sort by date
        statusBarChange(true);
        title = resultBody.props().movies[0].title;
        expect(title).toEqual("Kill bill v.1");
    });
});