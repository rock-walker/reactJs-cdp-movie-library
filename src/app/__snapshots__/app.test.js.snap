// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render app 1`] = `
<div>
  <header
    className="App-header"
    style={
      Object {
        "backgroundImage": "url(test-file-stub)",
      }
    }
  >
    <h3
      alt="logo"
      className="App-logo"
    >
      netflixroulette
    </h3>
    <div>
      <h1
        className="App-title"
      >
        Find your movie
      </h1>
      <input
        className="searchBar"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="searchby"
    >
      <span
        className="title"
      >
        search by
      </span>
      <label
        className="sTitle"
      >
        <input
          name="toggle"
          onClick={[Function]}
          type="radio"
        />
        <span>
          title
        </span>
      </label>
      <label
        className="sGenre"
      >
        <input
          name="toggle"
          onClick={[Function]}
          type="radio"
        />
        <span>
          genre
        </span>
      </label>
      <div>
        <input
          className="searchButton"
          onClick={[Function]}
          type="button"
          value="SEARCH"
        />
      </div>
    </div>
  </header>
  <div
    className="statusBar"
  >
    <span>
       
      4
       movies found 
    </span>
    <div
      className="searchby"
    >
      <span
        className="title"
      >
        Sort by
      </span>
      <label>
        <input
          id="date"
          name="sorting"
          onClick={[Function]}
          type="radio"
          value="date"
        />
        <label
          htmlFor="date"
        >
          release date
        </label>
      </label>
      <label>
        <input
          id="rate"
          name="sorting"
          onClick={[Function]}
          type="radio"
          value="rate"
        />
        <label
          htmlFor="rate"
        >
          rating
        </label>
      </label>
    </div>
  </div>
  <div
    className="resultBody"
  >
    <div
      className="movieItem"
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movieDescription"
      >
        <span
          className="title"
        >
          Kill bill v.1
        </span>
        <div
          className="year"
        >
          2004
        </div>
        <p>
          Action & Adventure
        </p>
      </div>
    </div>
    <div
      className="movieItem"
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movieDescription"
      >
        <span
          className="title"
        >
          Kill bill v.2
        </span>
        <div
          className="year"
        >
          2003
        </div>
        <p>
          Action & Adventure
        </p>
      </div>
    </div>
    <div
      className="movieItem"
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movieDescription"
      >
        <span
          className="title"
        >
          Pulp fiction
        </span>
        <div
          className="year"
        >
          1994
        </div>
        <p>
          Oscar-winning movies
        </p>
      </div>
    </div>
    <div
      className="movieItem"
      onClick={[Function]}
    >
      <img
        src="test-file-stub"
      />
      <div
        className="movieDescription"
      >
        <span
          className="title"
        >
          Jackie Brown
        </span>
        <div
          className="year"
        >
          1997
        </div>
        <p>
          Dramas
        </p>
      </div>
    </div>
  </div>
  <footer>
    <h3
      className="App-logo"
    >
      netflixroulette
    </h3>
  </footer>
</div>
`;
